code/modules/events/ninja/energy_katana.dm

```diff
@@ -0,0 +1,81 @@
+
+/obj/item/weapon/katana/energy
+	name = "energy katana"
+	desc = "A katana infused with strong energy."
+	icon_state = "energy_katana"
+	item_state = "energy_katana"
+	force = 40
+	throwforce = 20
+	armour_penetration = 15
+	var/datum/effect_system/spark_spread/spark_system
+
+/obj/item/weapon/katana/energy/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	if(!user || !target)
+		return
+
+	if(proximity_flag)
+		if(isobj(target) || issilicon(target))
+			spark_system.start()
+			playsound(user, "sparks", 50, 1)
+			playsound(user, 'sound/weapons/blade1.ogg', 50, 1)
+			target.emag_act(user)
+
+
+//If we hit the Ninja who owns this Katana, they catch it.
+//Works for if the Ninja throws it or it throws itself or someone tries
+//To throw it at the ninja
+/obj/item/weapon/katana/energy/throw_impact(atom/hit_atom)
+	if(ishuman(hit_atom))
+		var/mob/living/carbon/human/H = hit_atom
+		if(istype(H.wear_suit, /obj/item/clothing/suit/space/space_ninja))
```

Comment:
Wouldn't that be equally roundabout when it comes to people killing the ninja and wearing his suit?